exasol/ai/mcp/server/main.py

```diff
@@ -216,33 +254,110 @@ def get_mcp_settings() -> McpServerSettings:
 
 
 def create_mcp_server(
-    connection: DbConnection, config: McpServerSettings
+    connection: DbConnection, config: McpServerSettings, **kwargs
 ) -> ExasolMCPServer:
     """
     Creates the Exasol MCP Server and registers its tools.
     """
-    mcp_server = ExasolMCPServer(connection=connection, config=config)
+    mcp_server = ExasolMCPServer(connection=connection, config=config, **kwargs)
     register_tools(mcp_server, config)
     return mcp_server
 
 
-def main():
+def get_auth_provider(env: dict[str:Any]) -> AuthProvider | None:
     """
-    Main entry point that creates and runs the MCP server.
+    Creates one of the FastMCP OAuth2 providers, depending on content of the supplied
+    configuration dictionary. Currently, the dictionary is filled from the environment
+    variables. Going forward, configuration parameters will be kept in the NBC secret
+    store.
     """
```

Comment:
I suggest we don't focus too much on this function. Currently, it creates one of the 3 generic AuthProvides, each with a minimal set of parameters. This is required for running the integration tests. In a subsequent PR, I will add more options, including classes for specific Auth providers included in the FastMCP Authentication package. This will probably take some restructuring of the code. Maybe moving it into a separate file.